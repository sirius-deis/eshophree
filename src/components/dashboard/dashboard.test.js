import { render, screen } from "@testing-library/react";
import Dashboard from "./dashboard";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([])

describe("Dashboard component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({user: {user: {name: "John"}}})
  })

  it('should match snapshot', () => {
    const { container } = render(<Provider store={store}><Dashboard /></Provider>);
    expect(container).toMatchSnapshot();
  })

  it("should render dashboard component", () => {
    render(<Provider store={store}><Dashboard /></Provider>)
    expect(screen.getByText(/john/i)).toBeInTheDocument()
  })
})