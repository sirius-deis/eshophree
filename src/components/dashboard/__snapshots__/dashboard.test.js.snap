// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should match snapshot 1`] = `
<div>
  <div
    class="sc-bgqQcB cHXazT"
  >
    <h2>
      Hello, 
      <span
        class="sc-gTRrQi cTcuAO"
      >
        John
      </span>
    </h2>
    <p>
      From your account dashBoard you can view your recent orders, manage your shipping and billing addresses and edit your password and account details
    </p>
  </div>
</div>
`;
